Source/SwiftLintFramework/Rules/ColonRule.swift

```diff
@@ -161,3 +208,89 @@ public struct ColonRule: CorrectableRule, ConfigurationProviderRule {
         }
     }
 }
+
+// MARK: Dictionary Colon Rule
+extension ColonRule {
+
+    public func validateFile(_ file: File, kind: SwiftExpressionKind,
+                             dictionary: [String : SourceKitRepresentable]) -> [StyleViolation] {
+
+        let ranges = violationRangesInFile(file, kind: kind, dictionary: dictionary)
+        return ranges.map {
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severityConfiguration.severity,
+                           location: Location(file: file, byteOffset: $0.location))
+        }
+    }
+
+    private func violationRangesInFile(_ file: File, kind: SwiftExpressionKind,
+                                       dictionary: [String : SourceKitRepresentable]) -> [NSRange] {
+        guard kind == .dictionary,
+            let ranges = colonRanges(dictionary) else {
+                return []
+        }
+
+        let contents = file.contents.bridge()
+        return ranges.filter {
+            guard let colon = contents.substringWithByteRange(start: $0.location,
+                                                              length: $0.length) else {
+                                                                return false
+            }
+
+            if configuration.flexibleRightSpacing {
+                let isCorrect = colon.hasPrefix(": ") || colon.hasPrefix(":\n")
+                return !isCorrect
+            }
+
+            return colon != ": " && !colon.hasPrefix(":\n")
+        }
+    }
+
+    fileprivate func dictionaryViolations(in file: File,
+                                          dictionary: [String: SourceKitRepresentable])
+                                                                                    -> [NSRange] {
```

Comment:
elsewhere in the codebase that we need to split lines at the return type, we align the return arrow vertically with the parameters:

```swift
fileprivate func dictionaryViolations(in file: File,
                                      dictionary: [String: SourceKitRepresentable])
                                      -> [NSRange] {
```